.github/workflows/self-update.yml

```
@@ -0,0 +1,38 @@
+name: self-update
+
+permissions: {}
+
+on:
+  workflow_dispatch:
+  schedule:
+    - cron: "0 0 * * *"
+
+jobs:
+  update:
+    runs-on: ubuntu-latest
```

You could make the job conditional, instead of doing a runtime check:

```suggestion
    runs-on: ubuntu-latest
    if: github.event.repository.fork == true
```